### PR TITLE
Smart Helm values merger

### PR DESCRIPTION
**Pull request by agrimmer**
Signed-off-by: agrimmer <andreas.grimmer@dynatrace.com>

### CODE REVIEW

**Comment by christian-kreuzberger-dtx** [comment]
I think this PR shows how the merger should work.

Now that we have this feature, is there a chance we can get rid of the [replica_count_manipulator.go](https://github.com/keptn/keptn/blob/master/helm-service/pkg/configurationchanger/replica_count_manipulator.go)?

File: cli/internal/textutils.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"fmt"
+	"strings"
+)
+
+type node interface {
+	String() string
+}
+
+func newInnerNode() innerNode {
+	return innerNode{childs: make(map[string]node)}
```
Comment (jsoref) [inline cli/internal/textutils.go:13]: Generally you want `children`

**Review by jsoref — CHANGES_REQUESTED** [review]
Please add the other words the bot suggested into `expect.txt`, if you're on Linux, it has a command you can run to do it for you.

**Comment by warber** [comment]
closes #3341

File: cli/internal/textutils.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+func newInnerNode() *innerNode {
+	return &innerNode{
+		sortedKeys: []string{},
+		childs:     make(map[string]node),
+	}
+}
+
+type innerNode struct {
+	sortedKeys []string
+	childs     map[string]node
```
Comment (bacherfl) [inline cli/internal/textutils.go:21]: can we rename this to `children`?

**Review by bacherfl — APPROVED** [review]

**Pull request merged**